core/src/cluster_client.rs

```diff
@@ -1,6 +1,6 @@
 use crate::cluster_info::FULLNODE_PORT_RANGE;
```

Comment:
Can we move this entire file into `client/`

Reply:
Can I move ContactInfo into sdk?

Reply:
(FULLNODE_PORT_RANGE can just be passed in, I think.)

Reply:
hmm, moving ContactInfo into `sdk/` doesn't sound great.  I guess we'd need to pass in the `rpc' and `tpu` sockets as well as the port change.  Ugh, nevermind.   

But I don't love the name `cluster_client.rs`, how about `mk_client.rs`.

Also `mk_client_with_timeout()` only has one user, `storage_stage.rs` -- @sakridge, do we really need the custom timeout?  It would be nice to boil this file down to just `mk_client()`

Reply:
ThinClient and ThinClient::default()?

Reply:
I'd like to add a LightClient to this crate at some point too.

Reply:
Purging the `mk_` prefix would be good too. We use `create_` everywhere else in the codebase.  As for `mk_client.rs`,  that `mk` stands for `make` and there's no precedent for modules that start with a verb. They're all nouns.

Reply:
`mk_` prefix purged; moved module to solana-client
This version does pass in `rpc` and `tpu` sockets, which I know you weren't excited about @mvines , but I added a `client_facing_addr()` method to make this less titchy for users.
Thoughts?